csharp/src/Apache.Arrow.Adbc/AdbcDatabase11.cs

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Apache.Arrow.Adbc
+{
+    /// <summary>
+    /// Clients first initialize a database, then create a connection.
+    /// This gives the implementation a place to initialize and own any
+    /// common connection state.
+    /// For example, in-memory databases can place ownership of the actual
+    /// database in this object.
+    /// </summary>
+    public abstract class AdbcDatabase11 : IDisposable
```

Comment:
This class _must_ inherit from `AdbcDatabase` - if we want it to be returned from `AdbcDriver11`.

Do you think it would be helpful to have interface definitions for the different ADBC versions?

Reply:
I considered using interfaces as markers to indicate the supported version of ADBC, but in my experience C# isn't really designed in a way to make that a helpful approach.

Reply:
I've gone ahead and fixed all the "11" classes to return the correct "11" types.